xrpl/models/requests/request.py

```diff
@@ -106,6 +108,13 @@ class Request(BaseModel):
 
     id: Optional[Union[str, int]] = None
 
+    api_version: int = DEFAULT_API_VERSION
```

Comment:
There should be an additional test for another request that makes sure that the `api_version` field is handled properly - perhaps `account_info` since there is a significant difference in the shape of the output.

Reply:
Agree with this

Reply:
There should be an integration test that ensures that including an `api_version` param in a request type that is not `GenericRequest` still works correctly and properly returns the correct API version.

Reply:
Doesn't a unit test suffice?

It is rippled's responsibility to return the correct `api_version` response, why should the client libraries test the response?

Even in the case of `GenericRequest`, I'm only forwarding the `api_version` input specified by the user. The integration test pertaining to `tx_history` tests the older versions of rippled. I have not tested the validity of responses of rippled

Reply:
> Doesn't a unit test suffice?

In this case, I'd like to ensure that it matches rippled compatibility, so an integration test is needed.

> It is rippled's responsibility to return the correct `api_version` response, why should the client libraries test the response?

I'm not suggesting a test that actually fully tests the rippled response, but a test that ensures that when I put in `api_version: 1` in my request model, it actually gives me the rippled API v1 response and not the rippled API v2 response (in which case there's something wrong with the implementation of `api_version` in the xrpl-py model).

Reply:
In this PR, every `Request` will always specify an `api_version` field. It is mandatory for all the requests. 

If a `Request` omits a `api_version` field, then rippled returns v1 Response -- [docs](https://xrpl.org/docs/references/http-websocket-apis/api-conventions/request-formatting/#websocket-format)

Having said that, I can manually construct a JSON and validate API version of the `Response` object. There would be three cases:
- Request specifying  `api_version : 1`
- Request specifying  `api_version : 2`
- Request not specifying  `api_version` (through a dictionary object)

Reply:
Don't manually construct a JSON, that's not useful for testing the library. Construct a request model through the proper process.

Reply:
added tests in ae5b24f